Samples/StorageProviders/StorageProviders/MongoDBStorageProvider.cs

```diff
@@ -103,8 +99,9 @@ public Task<string> Read(string collectionName, string key)
             if (collection == null)
                 return Task.FromResult<string>(null);
 
-            var query = Query.EQ("key", key);
-            var existing = collection.FindOne(query);
+            var builder = Builders<BsonDocument>.Filter.Eq("key", key);
+
+            var existing = collection.Find(builder).FirstOrDefault();
```

Comment:
aren't there async methods to find? Any blocking I/O code is extremely bad for an orleans app